### PR TITLE
Fix warnings from new liveview

### LINKED ISSUE #272: Fix test compiler warnings
`mix test` spews warnings like so:

```
21:01:52.442 [info] Migrations already up
................*.......warning: Passing an atom to "for" in the form component is deprecated.
Instead of:

    <.form :let={f} for={:duplicate_form} ...>

You might do:

    <.form :let={f} for={%{}} as={:duplicate_form} ...>

Or, if you prefer, use to_form to create a form in your LiveView:

    assign(socket, form: to_form(%{}, as: :duplicate_form))

and then use it in your templates (no :let required):

    <.form for={@form}>

  (phoenix_live_view 0.20.3) lib/phoenix_component.ex:1489: Phoenix.Component.to_form/2
  (phoenix_live_view 0.20.3) lib/phoenix_component.ex:2213: Phoenix.Component."form (overridable 1)"/1
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/tag_engine.ex:68: Phoenix.LiveView.TagEngine.component/3
  (bike_brigade 0.1.0) lib/bike_brigade_web/core_components.ex:798: anonymous fn/2 in BikeBrigadeWeb.CoreComponents."simple_form (overridable 1)"/1
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:517: Phoenix.LiveView.Diff.invoke_dynamic/2
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:391: Phoenix.LiveView.Diff.traverse/7
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:544: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/7
  (elixir 1.14.2) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:389: Phoenix.LiveView.Diff.traverse/7
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:765: Phoenix.LiveView.Diff.render_component/9
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:705: Phoenix.LiveView.Diff.zip_components/5
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:690: anonymous fn/4 in Phoenix.LiveView.Diff.render_pending_components/6
  (stdlib 4.0.1) maps.erl:411: :maps.fold_1/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:635: Phoenix.LiveView.Diff.render_pending_components/6
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:143: Phoenix.LiveView.Diff.render/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/channel.ex:936: anonymous fn/4 in Phoenix.LiveView.Channel.render_diff/3
  (telemetry 1.0.0) /Users/tees/development/tees/dispatch/deps/telemetry/src/telemetry.erl:293: :telemetry.span/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/channel.ex:931: Phoenix.LiveView.Channel.render_diff/3

warning: inputs_for/3 without an anonymous function is deprecated. If you are using HEEx templates, use the new Phoenix.Component.inputs_for/1 component
  (phoenix_html 3.3.3) lib/phoenix_html/form.ex:620: Phoenix.HTML.Form.inputs_for/3
  (bike_brigade 0.1.0) lib/bike_brigade_web/live/campaign_live/duplicate_campaign_component.ex:22: anonymous fn/3 in BikeBrigadeWeb.CampaignLive.DuplicateCampaignComponent.render/1
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:391: Phoenix.LiveView.Diff.traverse/7
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:544: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/7
  (elixir 1.14.2) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:389: Phoenix.LiveView.Diff.traverse/7
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:544: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/7
  (elixir 1.14.2) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:389: Phoenix.LiveView.Diff.traverse/7
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:544: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/7
  (elixir 1.14.2) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:389: Phoenix.LiveView.Diff.traverse/7
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:544: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/7
  (elixir 1.14.2) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:389: Phoenix.LiveView.Diff.traverse/7
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:765: Phoenix.LiveView.Diff.render_component/9
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:705: Phoenix.LiveView.Diff.zip_components/5
  (phoenix_live_view 0.20.3) lib/phoenix_live_view/diff.ex:690: anonymous fn/4 in Phoenix.LiveView.Diff.render_pending_components/6

```

Most of them are the same: `warning: inputs_for/3 ` one. The `warning: Passing an atom to "for" in the form component is deprecated.` has been there for a while, but only shows up once.

### PR DESCRIPTION
**Pull request by mveytsman**
Closes #272

### CODE REVIEW

File: lib/bike_brigade_web/live/program_live/form_component.html.heex [hunk 1]
```diff
@@ -9,149 +9,157 @@
...
+            </:actions>
+          </.header>
 
-        <.table id="items" rows={inputs_for(f, :items) |> Enum.with_index()}>
-          <:col :let={{item_form, _i}} label="Name">
-            <.input type="hidden" field={{item_form, :id}} />
-            <.input type="hidden" field={{item_form, :program_id}} />
-            <.input type="text" field={{item_form, :name}} />
-          </:col>
+          <.table id="items" rows={f.impl.to_form(f.source, f, :items, [])}>
```
Comment (mveytsman) [inline lib/bike_brigade_web/live/program_live/form_component.html.heex:89]: Ooof I am not happy with this, but can't think of a better way

Comment (teesloane) [reply lib/bike_brigade_web/live/program_live/form_component.html.heex:89]: Curious, where in the docs did you find the `.impl.to_form` bit? I was looking around for how to solve this at one point and didn't find anything.

**Comment by teesloane** [comment]
CleanShot 2024-02-17 at 15 41 40@2x

Looking good.

**Review by teesloane — APPROVED** [review]

File: lib/bike_brigade_web/live/program_live/form_component.html.heex (hunk 1 above)
Comment (mveytsman) [reply lib/bike_brigade_web/live/program_live/form_component.html.heex]: Thanks for calling this out!

I found the `impl.to_form` by reading the code for `inputs_for`. But this wasn't the right way to do it - calling an impl directly is probably a bad idea, instead I switched to calling via the protocol `Phoenix.HTML.FormData.to_form/4`

**Pull request merged**